api-test/internal/report/html.go

```
@@ -0,0 +1,1260 @@
+	// unredacted for local debugging. Callers pass false unless the operator opted
+	// in; see config.Run.DebugShowSecrets for the boundary of what it affects. It
+	// never unmasks ConfigJSON/PlanConfigs, which their callers mask regardless.
+	ShowSecrets bool
+	Results     []result.ModuleResult
+}
+
+// Write emits <dir>/<plan>_<provider>_<ts>_t-<n>_p-<n>_f-<n>_sk-<n>_ki-<n>.{html,json}
+// and returns the HTML path.
+func Write(o Options) (string, error) {
+	dir := o.Dir
+	if err := os.MkdirAll(dir, 0o755); err != nil {
```

golangci-lint (2.12.2)

[medium] G301: Expect directory permissions to be 0750 or less

(gosec)